modules/git/repo_language_stats_gogit.go

```diff
@@ -45,6 +45,7 @@ func (repo *Repository) GetLanguageStats(commitID string) (map[string]int64, err
 	defer deferable()
 
 	sizes := make(map[string]int64)
+	keepLanguage := make(map[string]struct{})
```

Comment:
If I see that correctly, you simply want a constant time check if you can keep languages, right?
Why not simplify it to `map[string]bool` then, which is both clearer and more performant?